traffic_router/connector/src/main/java/com/comcast/cdn/traffic_control/traffic_router/secure/CertificateDataConverter.java

```diff
@@ -49,6 +81,76 @@ public HandshakeData toHandshakeData(final CertificateData certificateData) {
 		return null;
 	}
 
+	public boolean verifySubject(final X509Certificate certificate, final String hostAlias ) {
+		final String host = certificate.getSubjectDN().getName();
+		if (hostCompare(hostAlias,host)) {
+			return true;
+		}
+
+		try {
+			// This approach is probably the only one that is JDK independent
+			if (certificate.getSubjectAlternativeNames() != null) {
+				for (final List<?> altName : certificate.getSubjectAlternativeNames()) {
+					if (hostCompare(hostAlias, (String) altName.get(1))) {
+						return true;
+					}
+				}
+			}
+		}
+		catch (Exception e) {
+			log.error("Encountered an error while validating the certificate subject for service: "+hostAlias+", " +
+					"error: "+e.getClass().getSimpleName()+": " + e.getMessage(), e);
+			return false;
+		}
+
+		return false;
+	}
+
+	private boolean hostCompare(final String hostAlias, final String subject) {
+		if (hostAlias.contains(subject) || subject.contains(hostAlias)) {
+			return true;
+		}
+		final String[] chopped = subject.split("CN=", 2);
+		if (chopped != null && chopped.length > 1) {
+			String chop = chopped[1];
+			chop = chop.replaceFirst("\\*\\.", ".");
+			chop = chop.split(",", 2)[0];
+			if (chop.length()>0 && (hostAlias.contains(chop) || chop.contains(hostAlias))) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public boolean verifyModulus(final PrivateKey privateKey, final X509Certificate certificate) {
+		BigInteger privModulus = null;
+		if (privateKey instanceof BCRSAPrivateCrtKey) {
+			privModulus = ((BCRSAPrivateCrtKey) privateKey).getModulus();
+		} else if (privateKey instanceof RSAPrivateCrtKeyImpl) {
```

Comment:
Does this if-else chain need a default case? Otherwise, `privModulus` could end up being null and throw a NPE on L148, right? I suppose that would be captured on L76, but maybe it should be handled with a better error message here?

Reply:
Yes. I corrected it in the latest commit.